### PR TITLE
feat(mesh): add 6.1.1 sponsor record page

### PR DESCRIPTION
**Pull request by tyc67**
## Notable Change

- add sponsor record page `/point/sponsorship` [figma](https://www.figma.com/design/wqODc69zqgbu7TZFtQAmnN/2024_READr-Mesh_Web?node-id=8999-172407&m=dev)
- refactor `/point` page to integrate with cms data, implement infinite scroll and adjusted UI [figma](https://www.figma.com/design/wqODc69zqgbu7TZFtQAmnN/2024_READr-Mesh_Web?node-id=4526-213608&t=E8EUaM5vaqB41WMJ-4)

Screenshot 2024-08-14 at 11 56 36 AM
Screenshot 2024-08-14 at 11 49 53 AM
Screenshot 2024-08-14 at 11 50 43 AM
Screenshot 2024-08-14 at 11 56 17 AM
Screenshot 2024-08-14 at 11 55 01 AM
Screenshot 2024-08-14 at 11 54 29 AM

https://github.com/user-attachments/assets/352faefc-6593-47cb-b00b-9682217c3b8b

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1207975510192857

### CODE REVIEW

File: packages/mesh/app/point/sponsorship/page.tsx [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+import { notFound } from 'next/navigation'
+import { Fragment } from 'react'
+
+import { getCurrentUser } from '@/app/actions/auth'
+import { getMemberSponsorRecord } from '@/app/actions/sponsorship'
+import Icon from '@/components/icon'
+import PublisherDonateButton from '@/components/publisher-card/donate-button'
+
+export default async function Page() {
+  const user = await getCurrentUser()
```
Comment (erase2004) [inline packages/mesh/app/point/sponsorship/page.tsx:12]: 這裡應該用 UserContext

File: packages/mesh/graphql/query/member.gql [hunk 2]
```diff
@@ -356,3 +356,76 @@ query GetMemberFollowerList($customId: String, $take: Int!) {
...
+        charge
+        unlockSingle
+      }
+    }
+  }
+  transactions(
+    where: {
+      member: { id: { equals: $memberId } }
+      status: { equals: Success }
+      unlockStory: { id: { gte: 0 } }
```
Comment (erase2004) [inline packages/mesh/graphql/query/member.gql:430]: 需要 `active:  { equals: true }` 的條件，另外 `take: 10000` 的用意？

File: packages/mesh/app/point/_components/mesh-point.tsx [hunk 3]
```diff
@@ -158,31 +49,47 @@ export default function MeshPoint() {
...
-                  //TODO: update fetching logic
-                  return transactionData
+                initialList={transactionData['combinedRecord']}
+                pageSize={transactionRecordTake}
+                amountOfElements={200}
+                hasMoreData={hasMoreData}
+                fetchMoreTransaction={async (pageIndex: number) => {
+                  if (!hasMoreData) return []
+                  const { combinedRecord } = await getMemberTransactionRecord(
+                    '292',
```
Comment (erase2004) [inline packages/mesh/app/point/_components/mesh-point.tsx]: 這個 id 應該替換成 user.memberId ?

File: packages/mesh/app/point/_components/mesh-point.tsx [hunk 4]
```diff
@@ -158,31 +49,47 @@ export default function MeshPoint() {
             <div className="flex flex-row items-center justify-center gap-4 pt-6 sm:gap-5 sm:self-end sm:pt-0">
               <div className="flex flex-col items-center gap-1 sm:flex-row">
                 <p className="profile-title-2 text-primary-700">
-                  {sponsoredCount}次
+                  {transactionData?.totalSponsorCount || 0}次
```
Comment (erase2004) [inline packages/mesh/app/point/_components/mesh-point.tsx]: Ln. 41 已經有
```TypeScript
  if (!transactionData) return <Spinner />
```
後面跟 transactionData 有關的使用應該不需要再加上 `?` ?

**Review by erase2004 — COMMENTED** [review]
1. 現在找出會員贊助媒體的關聯，是用 `equals: $publisherId` 的方式，可以調整為 `in: $$publisherIdList` 來減少查找次數，
又或者，其實只要找 sponsor 資料上，publisher 不為空的即可 `id: { $gte: 0 }`

File: packages/mesh/app/point/_components/mesh-point.tsx [hunk 5]
```diff
@@ -158,31 +49,47 @@ export default function MeshPoint() {
...
                 </p>
-                <p className="profile-subtitle text-primary-500">已贊助次數</p>
+                <Link href={'/point/sponsorship'}>
+                  <p className="profile-subtitle text-primary-500">
+                    已贊助次數
+                  </p>
+                </Link>
```
Comment (caesarWHLee) [inline packages/mesh/app/point/_components/mesh-point.tsx:58]: <Link> 應該包含前面的次數？

File: packages/mesh/graphql/query/member.gql (hunk 2 above)
Comment (tyc67) [reply packages/mesh/graphql/query/member.gql:430]: 修改於 [c62542b](https://github.com/readr-media/Sachiel/pull/828/commits/c62542b54b376108cce54c6c742ca5844d83505b)

File: packages/mesh/app/point/_components/mesh-point.tsx (hunk 3 above)
Comment (tyc67) [reply packages/mesh/app/point/_components/mesh-point.tsx]: 修改於 [c62542b](https://github.com/readr-media/Sachiel/pull/828/commits/c62542b54b376108cce54c6c742ca5844d83505b)

File: packages/mesh/app/point/_components/mesh-point.tsx (hunk 4 above)
Comment (tyc67) [reply packages/mesh/app/point/_components/mesh-point.tsx]: 修改於 [c62542b](https://github.com/readr-media/Sachiel/pull/828/commits/c62542b54b376108cce54c6c742ca5844d83505b)

File: packages/mesh/app/point/_components/mesh-point.tsx (hunk 5 above)
Comment (tyc67) [reply packages/mesh/app/point/_components/mesh-point.tsx:58]: 修改於 [c62542b](https://github.com/readr-media/Sachiel/pull/828/commits/c62542b54b376108cce54c6c742ca5844d83505b)

File: packages/mesh/app/point/sponsorship/page.tsx (hunk 1 above)
Comment (tyc67) [reply packages/mesh/app/point/sponsorship/page.tsx:12]: 目前這裡是採用 server component render

File: packages/mesh/graphql/query/member.gql [hunk 6]
```diff
@@ -356,3 +356,80 @@ query GetMemberFollowerList($customId: String, $take: Int!) {
...
+        id
+        title
+      }
+    }
+  }
+}
+
+query GetMemberTransactions($memberId: ID!, $take: Int) {
+  member(where: { id: $memberId }) {
+    sponsorCount
```
Comment (erase2004) [inline packages/mesh/graphql/query/member.gql]: 這裡應該要加上 `where: { status: { equals: Success } }` ?，總贊助次數只有考慮成功的結果？

Comment (tyc67) [reply packages/mesh/graphql/query/member.gql]: 感謝提醒, 修改於 [a7ba347](https://github.com/readr-media/Sachiel/pull/828/commits/a7ba347e51b24d5d70b56fd755df1005f23186be)

**Review by caesarWHLee — APPROVED** [review]
LGTM 👍

**Review by erase2004 — APPROVED** [review]
LGTM 🥇

**Pull request merged**